core/src/wallet/common/synchronizers/AbstractBlockchainExplorerAccountSynchronizer.h

```diff
@@ -290,7 +290,7 @@ namespace ledger {
                     //But we may want to force sync of accounts within KEYCHAIN_OBSERVABLE_RANGE
                     auto discoveredAddresses = currentBatchIndex * buddy->halfBatchSize;
                     auto lastDiscoverableAddress = buddy->configuration->getInt(api::Configuration::KEYCHAIN_OBSERVABLE_RANGE).value_or(buddy->halfBatchSize);
-                    if (!isETHWallet && (!done || (done && hadTransactions) || lastDiscoverableAddress > discoveredAddresses)) {
+                    if (hasMultipleAddresses && (!done || (done && hadTransactions) || lastDiscoverableAddress > discoveredAddresses)) {
```

Comment:
Because you changed this line, I guess it’s time to clean it, especially according to Boole logic!

```
!done || (done && hadTransactions) == !done || hadTransactions
```

😉

Reply:
It should explicitly stay like it is because it is straightforward to understand each condition which leads to trigger the next synchro.